test/HubPool.Fixture.ts

```diff
@@ -44,7 +44,23 @@ export const hubPoolFixture = hre.deployments.createFixture(async ({ ethers }) =
     parentFixtureOutput.timer.address
   );
 
-  return { weth, usdc, dai, hubPool, ...parentFixtureOutput };
+  // Deploy a mock chain adapter and add it as the chainAdapter for the test chainId. Set the SpokePool to address 0.
+  const mockAdapter = await (await getContractFactory("Mock_Adapter", signer)).deploy();
+  await mockAdapter.transferOwnership(hubPool.address);
+  const mockSpoke = await (
+    await getContractFactory("MockSpokePool", signer)
+  ).deploy(weth.address, 0, parentFixtureOutput.timer.address);
+  await hubPool.setCrossChainContracts(repaymentChainId, mockAdapter.address, mockSpoke.address);
+
+  // Deploy mock l2 tokens for each token created before and whitelist the routes.
```

Comment:
nit: is there any benefit to running all these async funcs in parallel? I know the test network is already super fast but we could possibly get even faster?

Reply:
I dont think really. I think optimising for readability is the most important thing in fixtures IMHO.